src/feeds/WstETHPriceFeed.sol

```diff
@@ -0,0 +1,201 @@
+                stEthUsdPrice,
+                startedAt,
+                updatedAt,
+                answeredInRound
+            ) = stEthToUsdFallbackOracle();
+        }
+
+        int256 wstEthUsdPrice = int256(stEthPerToken) * stEthUsdPrice / 10 ** 8;
+
+        return (roundId, wstEthUsdPrice, startedAt, updatedAt, answeredInRound);
+    }
+
+    /** 
+    @notice Retrieves the latest price for the wstETH token
+    @return price The latest price for the wstETH token
+    */
+    function latestAnswer() external view returns (int256) {
+        (, int256 price, , , ) = latestRoundData();
+        return price;
+    }
+
+    /**
+     * @notice Retrieves number of decimals for the wstETH price feed
+     * @return decimals The number of decimals for the wstETH price feed
+     */
+    function decimals() external pure returns (uint8) {
+        return 18;
+    }
+
+    /**
+     * @notice Checks if a given price is out of the boundaries defined in the Chainlink aggregator.
+     * @param price The price to be checked.
+     * @param feed The Chainlink feed to retrieve the boundary information from.
+     * @return bool Returns `true` if the price is out of bounds, otherwise `false`.
+     */
+    function isPriceOutOfBounds(
+        int price,
+        IChainlinkFeed feed
+    ) public view returns (bool) {
+        IAggregator aggregator = IAggregator(feed.aggregator());
+        int192 max = aggregator.maxAnswer();
+        int192 min = aggregator.minAnswer();
+        return (max <= price || min >= price);
+    }
+
+    /**
+     * @notice Fetches the ETH to USD price and the stETH/ETH to get stETH/USD price, adjusts the decimals to match Chainlink oracles.
+     * @dev It will return the roundId, startedAt, updatedAt and answeredInRound from the ETH/USD Chainlink price feed when both oracles are not stale,
+     * in which case updatedAt will be zero
+     * @return roundId The round ID of the ETH/USD Chainlink price feed
+     * @return stEthToUsdPrice The latest stETH price in USD computed from the ETH/USD and stETH/ETH feeds
+     * @return startedAt The timestamp when the latest round of ETH/USD Chainlink price feed started
+     * @return updatedAt The timestamp when the latest round of ETH/USD Chainlink price feed was updated
+     * @return answeredInRound The round ID of the ETH/USD Chainlink price feed in which the answer was computed
+     */
+    function stEthToUsdFallbackOracle()
+        public
+        view
+        returns (uint80, int256, uint256, uint256, uint80)
+    {
+        (
+            uint80 roundId,
+            int256 ethToUsdPrice,
+            uint256 startedAt,
+            uint256 updatedAt,
+            uint80 answeredInRound
+        ) = ethToUsd.latestRoundData();
+
+        (
+            ,
+            int256 stEthToEthPrice,
+            ,
+            uint256 updatedAtStEth,
+        ) = stEthToEth.latestRoundData();
+
+        int256 stEthToUsdPrice = ethToUsdPrice * stEthToEthPrice / 10**18;
```

Comment:
This answer is in terms of `1e8`, while the other is `1e18`.

I think you should divide by `1e8` instead of `1e18` in order to arrive at this oracle's defined 18 decimals.